doc/arch/adr-004-user-roles-workflow.md

```diff
@@ -0,0 +1,78 @@
+        Draft --> Submitted: Submit
+        NR --> Draft: Respond
+        Submitted --> IR: Review
+    }
+    state Validator {
+        IR --> NR: Request Changes
+        IR --> Approved: Approve
+        Approved --> IR: Unapprove
+    }
+    Approved --> [*]
+```
+
+A boundary can be in one of five stages:
+
+- **Draft**: A newly created boundary that the Contributor is still working on. Not ready for review. Can be _edited_ by Contributor, can be _viewed_ by Validator.
+- **Submitted**: A boundary ready for review, as submitted by a Contributor. Can be _viewed_ by Contributor. A Validator can begin reviewing it, which will move it to:
+- **In Review**: Under active review by a Validator. Can be _viewed_ by Contributor, can be _annotated_ by Validator.
+- **Needs Revisions**: The bounday has been reviewed by a Validator who is requesting revisisions. Can be _viewed_ by Validator. A Contributor can respond to it by creating a new **Draft**.
+- **Approved**: The boundary has been reviewed by a Validator and found satisfactory.
+
+## Permissions
+
+A user can interact with a boundary in one of three ways:
+
+- **View**: A user can view a boundary, look at its shape, turn data layers on and off, turn reference images on and off, adjust their opacity, turn the shape on and off, and select a basemap. This is a read-only interaction, where none of these adjustments get saved to the boundary.
+- **Annotate**: Just like **View**, with the additional ability to add notes to particular areas on the map. Used by Validators for reviewing.
+- **Edit**: Just like **View**, with the additional ability to change the shape, delete the shape, upload a new shape, add reference images, set their opacity, and have everything be saved to the boundary. Used by Contributors before they submit a boundary.
+
+Here is a table of permissions for different users at different stages of the boundary:
+
+|                   | Draft | Submitted    | In Review | Needs Revisions | Approved |
+|-------------------|-------|--------------|-----------|-----------------|----------|
+| **Contributor**   | Edit  | View         | View      | Create Draft    | View     |
+| **Validator**     | View  | Start Review | Annotate  | View            | View     |
+| **Administrator** | Edit  | Start Review | Annotate  | Create Draft    | View     |
+
+As can be seen, Administrators can work like both Contributors and Validators.
+
+## Visibility
+
+When a new Contributor is added to the system, they must be associated with one or more **Utilities**. A Contributor can submit a boundary for any Utility they are associated with. A Contributor can view all boundaries in any stage related to their Utilities.
+
+When a new Validator is added to the system, they must be associated with a **US State**. A Validator can view all boundaries at all stages for all Utilities related to their US State.
+
+An Administrator can view all boundaries at all stages in the system.
+
+## Notifications
```

Comment:
There are two additional notifications:
- Boundary submitted notification for _contributors_
- Boundary approved notification for contributors